pkg/gather/clusterconfig/clusterconfig_test.go

```diff
@@ -88,6 +92,147 @@ func TestConfigMapAnonymizer(t *testing.T) {
+		inputObj   *imageregistryv1.Config
+		evalOutput func(t *testing.T, obj *imageregistryv1.Config)
+	}{
+		{
+			name: "httpSecret",
+			inputObj: &imageregistryv1.Config{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "cluster",
+				},
+				Spec: imageregistryv1.ImageRegistrySpec{
+					HTTPSecret: "secret",
+				},
+			},
+			evalOutput: func(t *testing.T, obj *imageregistryv1.Config) {
+				if obj.Spec.HTTPSecret != "xxxxxx" {
+					t.Errorf("expected HTTPSecret anonymyzed, got %q", obj.Spec.HTTPSecret)
+				}
+			},
+		},
+		{
+			name: "s3",
+			inputObj: &imageregistryv1.Config{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "cluster",
+				},
+				Spec: imageregistryv1.ImageRegistrySpec{
+					Storage: imageregistryv1.ImageRegistryConfigStorage{
+						S3: &imageregistryv1.ImageRegistryConfigStorageS3{
+							Bucket:         "foo",
+							Region:         "bar",
+							RegionEndpoint: "point",
+							KeyID:          "key",
+						},
+					},
+				},
+			},
+			evalOutput: func(t *testing.T, obj *imageregistryv1.Config) {
+				if obj.Spec.Storage.S3.Bucket == "foo" {
+					t.Errorf("expected s3 bucket anonymyzed, got %q", obj.Spec.Storage.S3.Bucket)
+				}
+				if obj.Spec.Storage.S3.Region == "bar" {
+					t.Errorf("expected s3 region anonymyzed, got %q", obj.Spec.Storage.S3.Region)
+				}
+				if obj.Spec.Storage.S3.RegionEndpoint == "point" {
+					t.Errorf("expected s3 region endpoint anonymyzed, got %q", obj.Spec.Storage.S3.RegionEndpoint)
+				}
+				if obj.Spec.Storage.S3.KeyID == "key" {
+					t.Errorf("expected s3 keyID anonymyzed, got %q", obj.Spec.Storage.S3.KeyID)
+				}
+			},
+		},
+		{
+			name: "azure",
+			inputObj: &imageregistryv1.Config{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "cluster",
+				},
+				Spec: imageregistryv1.ImageRegistrySpec{
+					Storage: imageregistryv1.ImageRegistryConfigStorage{
+						Azure: &imageregistryv1.ImageRegistryConfigStorageAzure{
+							AccountName: "account",
+							Container:   "container",
+						},
+					},
+				},
+			},
+			evalOutput: func(t *testing.T, obj *imageregistryv1.Config) {
+				if obj.Spec.Storage.Azure.AccountName == "account" {
+					t.Errorf("expected azure account name anonymyzed, got %q", obj.Spec.Storage.Azure.AccountName)
+				}
+				if obj.Spec.Storage.Azure.Container == "container" {
+					t.Errorf("expected azure container anonymyzed, got %q", obj.Spec.Storage.Azure.Container)
+				}
+			},
+		},
+		{
+			name: "gcs",
+			inputObj: &imageregistryv1.Config{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "cluster",
+				},
+				Spec: imageregistryv1.ImageRegistrySpec{
+					Storage: imageregistryv1.ImageRegistryConfigStorage{
+						GCS: &imageregistryv1.ImageRegistryConfigStorageGCS{
+							Bucket:    "bucket",
+							Region:    "region",
+							ProjectID: "foo",
+							KeyID:     "bar",
+						},
+					},
+				},
+			},
+			evalOutput: func(t *testing.T, obj *imageregistryv1.Config) {
+				if obj.Spec.Storage.GCS.Bucket == "foo" {
```

Comment:
`s/foo/bucket`, or maybe just checking it contains noething but `/x*/` would be better (and reusable for all keys to prevent this kind of copy/paste problems)

Reply:
it does not matter :-) this just checks the Bucket value was modified by anonymyzer.

Reply:
And if you read it carefully, you will find this test actually would catch even if the anonymizer was not there, becuase the strings are different (`foo` vs. `bucket`)

Reply:
updated